packages/coil-extension/src/background/services/BackgroundEvents.ts

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { EventEmitter } from 'events'
+
+import { inject, injectable } from 'inversify'
+import { WextApi } from '@webmonetization/wext/tokens'
+
+const events = {
+  'tabs.onRemoved': chrome.tabs.onRemoved,
+  'tabs.onReplaced': chrome.tabs.onReplaced,
+  'tabs.onActivated': chrome.tabs.onActivated,
+  'tabs.onActiveChanged': chrome.tabs.onActiveChanged,
+  'tabs.onSelectionChanged': chrome.tabs.onSelectionChanged,
+  'tabs.onCreated': chrome.tabs.onCreated,
+  'tabs.onAttached': chrome.tabs.onAttached,
+  'tabs.onDetached': chrome.tabs.onDetached,
+
+  // TODO: this will need special handling ...
```

Comment:
// Don't need to return true here, not using sendResponse
        // https://developer.chrome.com/apps/runtime#event-onMessage